lib/src/facts/linux/dmi_resolver.cc

```diff
@@ -44,6 +44,14 @@ namespace facter { namespace facts { namespace linux {
             return {};
         }
 
+        // Replace any non-printable ASCII characters with '.'
+        // This mimics the behavior of dmidecode
+        for (auto& c : value) {
+            if (c < 32 || c == 127) {
```

Comment:
The trim call needs to happen before this though, otherwise trailing newlines won't be trimmed (instead converted to '.').  Pushing up fix.